### PR TITLE
Add Adventure/MiniMessage as new message parser to acf

### PR DESCRIPTION
**Pull request by JOO200**
I added Adventure as a module to acf. It's still a work in progress.

TODOs:

- [x] Make a proper implementation for MessageType and MessageFormatters. (take a look at the picture below)
- [ ] Check every existing message key so there's no problem with weird color codes.

grafik

### CODE REVIEW

**Comment by JOO200** [comment]
I fixed issues with the MessageType and Formatter. I tested the implementation on bukkit and bungee and it works.

I'm not happy with the workaround for the formatter colors but I don't think there's another better solution without loosing backwarts compatibility.

To use the MiniMessage parser:

- Create a BukkitCommandManager or BungeeCommandManager
- enable the unstable api "adventure"
- have fun with MiniMessage.

I modified the example. e.g.
```java
        issuer.sendMessage("<red>This is a <yellow>" +
                "<hover:show_text:'<yellow>Click here'>" +
                "<click:open_url:'https://docs.adventure.kyori.net/minimessage.html'>clickable message</click></hover></yellow>.");
```
will send a message with formatted text.

I exposed the ACFBukkitAdventureManager and ACFBungeeAdventureManager classes (searching better class names) through the BukkitCommandManager and BungeeCommandManager to wrap CommandIssuers but I'm not sure if that's necessary.
You can simply call "XAudiences.create(plugin)" to receive the same (cached) Audience wrapper as they are cached by adventure.

File: bukkit/src/main/java/co/aikar/commands/BukkitCommandManager.java [hunk 1]
```diff
@@ -403,4 +414,19 @@ protected boolean handleUncaughtException(BaseCommand scope, RegisteredCommand r
         }
         return super.handleUncaughtException(scope, registeredCommand, sender, args, t);
     }
+
+    @Override
+    public void enableUnstableAPI(String api) {
+        super.enableUnstableAPI(api);
+
+        if ("adventure".equals(api) && adventureAvailable) {
+            log(LogLevel.INFO, "Enabling Adventure");
```
Comment (MiniDigger) [inline bukkit/src/main/java/co/aikar/commands/BukkitCommandManager.java:423]: not sure a message here is necessary, ppl already complain about the current messages...

**Comment by MiniDigger** [comment]
might be worse to expose the minimessage instance, so that you can provide your own one, for handling placeholders, custom transformations, whatever.

**Comment by JOO200** [comment]
MiniMessage instance is exposed and I added a context resolver for the Audience class for the command issuer.

I change the draft to a PR now as it's working for me.

**Comment by GeorgeV220** [comment]
Any updates on this?

**Comment by JOO200** [comment]
Short answer: No, there is no update here.
The used MiniMessage version is outdated and it should be updated to a never version.

I use ACF for my own server so I only need to support my platforms. That's why I created my own fork of ACF [here](https://github.com/JOO200/commands/tree/terraconia). I replaced the existing messaging stuff completly with adventure and minimessage and made many changes for my own server. It should compile without any issues but it only supports velocity and paper.

**Pull request closed without merging**